### PR TITLE
22 nov

### PR DESCRIPTION
**Pull request by rbramand-xilinx**
Added changes that support program and scan sub commands
Changes also include query_device partial Implementation to provide idea of implementation

### CODE REVIEW

**Comment by maxzhen** [comment]
@rbramand-xilinx , can we think of a better name for the pull request?

**Comment by stsoe** [comment]
I will be reviewing this request and redoing some of the code.

File: src/runtime_src/core/include/xrt.h [hunk 1]
```diff
@@ -54,6 +54,7 @@
 #include "xcl_app_debug.h"
 #include "xclerr.h"
 #include "xclhal2_mem.h"
+#include "boost/any.hpp"
```
Comment (stsoe) [inline src/runtime_src/core/include/xrt.h]: Not a valid include in xrt.h

File: src/runtime_src/core/include/xrt.h [hunk 2]
```diff
@@ -264,7 +265,7 @@ XCL_DRIVER_DLLESPEC unsigned int xclProbe();
  *
  * Return:         Device handle
  */
-XCL_DRIVER_DLLESPEC xclDeviceHandle xclOpen(unsigned int deviceIndex, const char *logFileName,
+ xclDeviceHandle xclOpen(unsigned int deviceIndex, const char *logFileName,
```
Comment (stsoe) [inline src/runtime_src/core/include/xrt.h]: DLL export macros cannot be removed.  I will fix it.

**Comment by stsoe** [comment]
Obsolete

**Pull request closed without merging**